app/ydoc-shared/src/util/net/__tests__/YjsTransport.test.ts

```diff
@@ -0,0 +1,236 @@
+import type { IJSONRPCData } from '@open-rpc/client-js/build/Request.js'
+import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest'
+import * as Y from 'yjs'
+import { YjsTransport } from '../YjsTransport'
+
+// Helper function to create JSON-RPC notification data (no id = no response expected)
+function createNotification(method: string, params?: any): IJSONRPCData {
+  return {
+    internalID: Math.random(),
+    request: {
+      jsonrpc: '2.0',
+      method,
+      ...(params === undefined ? { params: null } : { params }),
+    },
+  }
+}
+
+// Polyfill DOM event types for Node.js environment
+if (typeof globalThis.CloseEvent === 'undefined') {
+  ;(globalThis as any).CloseEvent = class CloseEvent extends Event {
```

Comment:
Line prefixed by `;`?
```suggestion
  (globalThis as any).CloseEvent = class CloseEvent extends Event {
```